src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java

```diff
@@ -42,7 +42,7 @@
     private final boolean isSocksProxyDisabled;
 
     public QueryServiceConnection(String url, Properties properties) throws SQLException {
-        this.properties = properties; // fixme: do deeepCopy and modify the props
```

Comment:
Please modify setToken method to have reference to token at connection level. Otherwise token is not cached.